test/tests/other.test.ts

```diff
@@ -13,11 +13,13 @@ const files = {
 test('Can format an Astro file with frontmatter', files, 'other/frontmatter');
 
 test('Can format an Astro file with embedded JSX expressions', files, 'other/embedded-expr');
-test(
-	'Options are passed to other Prettier Plugins when parsing embedded JSX expressions',
-	files,
-	'other/embedded-expr-options'
-);
+
+// TODO: This doesn't seem to work in Prettier 3, hmm
+// test(
+// 	'Options are passed to other Prettier Plugins when parsing embedded JSX expressions',
+// 	files,
+// 	'other/embedded-expr-options'
+// );
```

Comment:
@thecrypticace Prettier 3 changed how they resolve parsers and it's not possible anymore to just pass a function, it needs to be a defined parser in the plugin (as far as I could tell)

I wasn't able to get this test to pass, do you have any ideas? See https://github.com/withastro/prettier-plugin-astro/blob/55849e6cbf44adaffe49d45b6a40dd18a4ce203e/src/index.ts for where the expression parser is defined now.

Reply:
Sorry just now seeing this. Github has been doing a bad job of delivering notifications lately (sometimes delayed, sometimes not at all).

I'm not sure at first glance but I'll take a look.

Reply:
No worries! I saw that Tailwind shipped support for Prettier 3 today, awesome work! 😄

Reply:
@Princesseuh here you go: https://github.com/thecrypticace/prettier-plugin-astro/commit/6720257575c336dc2a69ed84474c9b5ac8258d0f

Since Prettier no longer passes parsers through it's no longer possible to reference the `babel-ts` that's been overwritten by an earlier plugin at runtime without a direct reference to it.

This does raise an interesting point though. It probably means we'll have to handle this explicitly in the Tailwind plugin one we ship plugin support for the Prettier v3 version.

With that, should the expression parser be called something like `astroExpressionParser` or `__astro_expression` (kind like Prettier's builtin `__js_expression`)? I'm not sure if `__` is reserved or means something to Prettier. Just kinda thinking out loud.

Reply:
Ahh, thank you! I wasn't sure if it was okay to make it use the expression parser or if it still needed to use `babel-ts`. Glad to know it's okay and other plugins can handle it (even though it's unfortunate.)